ws-server/src/server.rs

```diff
@@ -819,8 +818,8 @@ impl<M> Builder<M> {
 	///
 	/// let builder = WsServerBuilder::default().set_id_provider(RandomStringIdProvider::new(16));
 	/// ```
-	pub fn set_id_provider(mut self, id_provider: impl IdProvider + 'static) -> Self {
```

Comment:
> If I understand right, the main motivation for the change was to remove this generic, which may otherwise have been propagated as some type T: IdProvider type at higher levels?

exactly, yes

I guess somebody could have passed a Box<dyn IdProvider> to this as it stood (maybe making sure we had an impl <T: IdProvider> IdProvider for Box<T> if needed). if the reason for the changes here that doing that would have meant we had a double boxing (a Box inside an Arc) and after the change we have a single Boxing?

that sounds better, lemme check if that works